ArduCopter/compass_cal.cpp

```diff
@@ -0,0 +1,18 @@
+/// -*- tab-width: 4; Mode: C++; c-basic-offset: 4; indent-tabs-mode: nil -*-
+
+#include "Copter.h"
+
+#define COMPASS_CAL_STICK_DELAY 2.0f // 2 seconds
+#define COMPASS_CAL_DELAY 5.0f
+
+void Copter::compass_cal_update() {
+    compass.compass_cal_update();
+    static bool cal_has_run = false;
```

Comment:
why is this state stored here? Can't the compass library do the reboot itself if the user has requested it?